File: benchmarks/primes.c
#include <stdio.h>
#include <stdlib.h>

int main(int argc, char **argv) {
    long long limit = argc > 1 ? atoll(argv[1]) : 10000000LL;
    unsigned char *primes = malloc((size_t)limit);
    if (!primes) return 1;

    for (long long i = 0; i < limit; i++) primes[i] = 1;
    primes[0] = 0;
    if (limit > 1) primes[1] = 0;

    for (long long p = 2; p * p < limit; p++) {
        if (primes[p]) {
            for (long long i = p * p; i < limit; i += p)
                primes[i] = 0;
        }
    }

    long long count = 0;
    for (long long i = 0; i < limit; i++)
        if (primes[i]) count++;

    printf("%lld\n", count);
    free(primes);
    return 0;
}

File: benchmarks/primes.go
package main

import (
    "fmt"
    "os"
    "strconv"
)

func main() {
    limit := 10000000
    if len(os.Args) > 1 {
        if v, err := strconv.Atoi(os.Args[1]); err == nil {
            limit = v
        }
    }

    primes := make([]byte, limit)
    for i := range primes {
        primes[i] = 1
    }
    if limit > 0 { primes[0] = 0 }
    if limit > 1 { primes[1] = 0 }

    for p := 2; p*p < limit; p++ {
        if primes[p] == 1 {
            for i := p * p; i < limit; i += p {
                primes[i] = 0
            }
        }
    }

    count := 0
    for i := 0; i < limit; i++ {
        if primes[i] == 1 {
            count++
        }
    }
    fmt.Println(count)
}